app/models/notification_group.rb

```diff
@@ -0,0 +1,44 @@
+class NotificationGroup
+  attr_reader :count, :notifications_hash
+
+  def initialize(notifications)
+    @notifications_hash = build_notifications_hash(notifications)
+    @count = notifications.count
+  end
+
+  def each(&block)
+    @notifications_hash.each(&block)
+  end
+
+  def raw_hash
+    @notifications_hash
+  end
+
+
+  private
+
+  # The resulting notifications hash has the format:
+  # {
+  #   project1 => [
+  #     [item1, [notifications]],
```

Comment:
How come you decided to go with nested arrays instead of hashes?
Not requesting a change, just wondering what the thought process was.

```
{
  project1 => [
    { item1 => [notifications] },
    { item2 => [...] }
  ],
  project2 => ...
}
```

Reply:
This is a side-effect of the group_by method:
```
> h = {1 => 'one', 2 => 'two', 3 => 'three'}
=> {1=>"one", 2=>"two", 3=>"three"}
> h.group_by { |k,v| k % 2 }
=> {1=>[[1, "one"], [3, "three"]], 0=>[[2, "two"]]}
```
note that the values are returned as arrays.

I do agree that the hash is more readable/semantic, but with the way that we iterate through this hash (see digest.html.erb) I don't think the format matters. I'd be happy to do some preprocessing of the notifications hash should the need arise in the future.